File: 30-days-of-code/day8/solution.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
  inputString += inputStdin;
});

process.stdin.on('end', _ => {
  inputString = inputString.replace(/\s*$/, '')
    .split('\n')
    .map(str => str.replace(/\s*$/, ''));

  main();
});

function readLine() {
  return inputString[currentLine++];
}

function main() {
  const n = parseInt(readLine());
  const phoneBook = {};
  for (let i = 0; i < n; i++) {
    const lineSplit = readLine().split(' ');
    const name = lineSplit[0];
    const number = lineSplit[1];
    phoneBook[name] = number;
  }

  while (currentLine < inputString.length) {
    const queryName = readLine();
    if (queryName in phoneBook) {
      console.log(queryName + '=' + phoneBook[queryName]);
    } else {
      console.log('Not found');
    }
  }
}

File: 30-days-of-code/day8/solution.py
if __name__ == '__main__':
    n = int(input())
    phone_book = {}
    for i in range(0, n):
        name, number = [x for x in input().split()]
        phone_book[name] = number

    while True:
        try:
            query_name = input()
            if query_name in phone_book:
                print(query_name + '=' + phone_book[query_name])
            else:
                print('Not found')
        except EOFError:
            break
